pkg/sdk/dynamic_table.go

```diff
@@ -0,0 +1,248 @@
+	RefreshMode         DynamicTableRefreshMode
+	RefreshModeReason   string
+	Warehouse           string
+	Comment             string
+	Text                string
+	AutomaticClustering bool
+	SchedulingState     DynamicTableSchedulingState
+	LastSuspendedOn     time.Time
+	IsClone             bool
+	IsReplica           bool
+	DataTimestamp       time.Time
+}
+
+func (dt *DynamicTable) ID() AccountObjectIdentifier {
+	return NewAccountObjectIdentifier(dt.Name)
+}
+
+type dynamicTableRow struct {
+	CreatedOn           time.Time      `db:"created_on"`
+	Name                string         `db:"name"`
+	Reserved            string         `db:"reserved"`
+	DatabaseName        string         `db:"database_name"`
+	SchemaName          string         `db:"schema_name"`
+	ClusterBy           string         `db:"cluster_by"`
+	Rows                int            `db:"rows"`
+	Bytes               int            `db:"bytes"`
+	Owner               string         `db:"owner"`
+	TargetLag           string         `db:"target_lag"`
+	RefreshMode         string         `db:"refresh_mode"`
+	RefreshModeReason   sql.NullString `db:"refresh_mode_reason"`
+	Warehouse           string         `db:"warehouse"`
+	Comment             string         `db:"comment"`
+	Text                string         `db:"text"`
+	AutomaticClustering string         `db:"automatic_clustering"`
+	SchedulingState     string         `db:"scheduling_state"`
+	LastSuspendedOn     sql.NullTime   `db:"last_suspended_on"`
+	IsClone             bool           `db:"is_clone"`
+	IsReplica           bool           `db:"is_replica"`
+	DataTimestamp       time.Time      `db:"data_timestamp"`
+}
+
+func (dtr *dynamicTableRow) toDynamicTable() *DynamicTable {
+	dt := &DynamicTable{
+		CreatedOn:           dtr.CreatedOn,
+		Name:                dtr.Name,
+		Reserved:            dtr.Reserved,
+		DatabaseName:        dtr.DatabaseName,
+		SchemaName:          dtr.SchemaName,
+		ClusterBy:           dtr.ClusterBy,
+		Rows:                dtr.Rows,
+		Bytes:               dtr.Bytes,
+		Owner:               dtr.Owner,
+		TargetLag:           dtr.TargetLag,
+		RefreshMode:         DynamicTableRefreshMode(dtr.RefreshMode),
+		Warehouse:           dtr.Warehouse,
+		Comment:             dtr.Comment,
+		Text:                dtr.Text,
+		AutomaticClustering: dtr.AutomaticClustering == "ON", // "ON" or "OFF
+		SchedulingState:     DynamicTableSchedulingState(dtr.SchedulingState),
+		IsClone:             dtr.IsClone,
+		IsReplica:           dtr.IsReplica,
+		DataTimestamp:       dtr.DataTimestamp,
+	}
+	if dtr.RefreshModeReason.Valid {
+		dt.RefreshModeReason = dtr.RefreshModeReason.String
+	}
+	if dtr.LastSuspendedOn.Valid {
+		dt.LastSuspendedOn = dtr.LastSuspendedOn.Time
+	}
+	return dt
+}
+
+func (dt *dynamicTables) Show(ctx context.Context, opts *ShowDynamicTableOptions) ([]*DynamicTable, error) {
+	if opts == nil {
+		opts = &ShowDynamicTableOptions{}
+	}
+	if err := opts.validate(); err != nil {
```

Comment:
Please consult e.g. `database_role_impl.go`. You can simplify code in this method by using our helper `validateAndQuery` and `convertRows` functions (the second one requires you to change `toDynamicTable` to `convert`).

Reply:
done